typescript/sdk/src/token/checker.ts

```diff
@@ -0,0 +1,90 @@
+import { BigNumber } from 'ethers';
+
+import { ERC20, ERC20__factory } from '@hyperlane-xyz/core';
+
+import { TokenMismatchViolation } from '../deploy/types';
+import { HyperlaneRouterChecker } from '../router/HyperlaneRouterChecker';
+import { ChainName } from '../types';
+
+import { HypERC20App } from './app';
+import {
+  ERC20RouterConfig,
+  HypERC20Config,
+  TokenMetadata,
+  isCollateralConfig,
+  isNativeConfig,
+  isSyntheticConfig,
+} from './config';
+import { HypERC20Factories } from './contracts';
+
+export class HypERC20Checker extends HyperlaneRouterChecker<
+  HypERC20Factories,
+  HypERC20App,
+  ERC20RouterConfig
+> {
+  async checkChain(chain: ChainName): Promise<void> {
+    await super.checkChain(chain);
+    await this.checkToken(chain);
+  }
+
+  async checkToken(chain: ChainName): Promise<void> {
+    const checkERC20 = async (
+      token: ERC20,
+      config: HypERC20Config | boolean, // either check for validity if config is present, or check for existence if config is boolean
```

Comment:
I recommend against using union types like this. It's fine when the fn's behavior won't vary (e.g. `add(value:string|number)`) but encoding behavioral differences like this feels fragile. WDYT?

Reply:
fixed